File: Data Structures & Algorithms/daily-temperatures/submission-4.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        
        
        l = 0 
        r = 1   
        ret = []

        for l in range (len (temperatures)):
            curr = temperatures [l]
            num = 0
            found = False
            for r in range (l, len (temperatures)):
                if temperatures[r] > curr: 
                    found = True
                    break

                num+=1

            if found is True:
                ret.append (num)
            else:
                ret.append (0)
        return ret

File: Data Structures & Algorithms/duplicate-integer/submission-4.go
func hasDuplicate(nums []int) bool {

    values := make(map[int]int)

    for _, num := range nums {

        value, ok := values [num]

        if !ok {
            // value not found
            values [num] = 0
        }
        
        values [num] = value + 1
        

    }

    for _, count := range values {
        if count > 1 {
            return true
        }
    }

    return false

}
